src/dev/ci_setup/checkout_sibling_es.sh

```diff
@@ -64,13 +64,15 @@ function checkout_sibling {
 
     function checkout_clone_target {
       pick_clone_target
-      if [[ $cloneBranch = "6.x"  && $cloneAuthor = "elastic" ]]; then
-        export TEST_ES_FROM=snapshot
+
+      if [[ "$cloneAuthor/$cloneBranch" != "elastic/$KIBANA_PKG_BRANCH" ]]; then
+        echo " -> Setting TEST_ES_FROM=source so that ES in tests will be built from $cloneAuthor/$cloneBranch"
+        export TEST_ES_FROM=source
```

Comment:
We also need to account for if we are checking out against the PR authors elasticsearch ES branch.

Reply:
This will trigger any time we are not checking out `elastic/master`, or `elastic/6.x`, or whatever the `branch` in package.json is, so that should be covered.